From 373cd5c53d5ea6622c319ecd84e29e2737d488bd Mon Sep 17 00:00:00 2001
From: Theodore Ts'o <tytso@mit.edu>
Date: Sun, 22 May 2011 16:12:35 -0400
Subject: [PATCH] ext4: don't show mount options in /proc/mounts if there is no journal
patch-mainline: 2.6.39
Reference: BF#95153

After creating an ext4 file system without a journal:

  # mke2fs -t ext4 -O ^has_journal /dev/sda
  # mount -t ext4 /dev/sda /test

the /proc/mounts will show:
"/dev/sda /test ext4 rw,relatime,user_xattr,acl,barrier=1,data=writeback 0 0"
which can fool users into thinking that the fs is using writeback mode.

So don't set the writeback option when the journal has not been
enabled; we don't depend on the writeback option being set, since
ext4_should_writeback_data() in ext4_jbd2.h tests to see if the
journal is not present before returning true.

Reported-by: Robin Dong <sanbai@taobao.com>
Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
Signed-off-by: Coly Li <bosong.ly@taobao.com>
---
 fs/ext4/super.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

Index: linux-2.6.32-220.0.0.el5/fs/ext4/super.c
===================================================================
--- linux-2.6.32-220.0.0.el5.orig/fs/ext4/super.c	2011-12-13 14:19:58.476436000 +0800
+++ linux-2.6.32-220.0.0.el5/fs/ext4/super.c	2011-12-13 14:21:44.608962278 +0800
@@ -3269,7 +3269,6 @@
 		goto failed_mount4;
 	} else {
 		clear_opt(sbi->s_mount_opt, DATA_FLAGS);
-		set_opt(sbi->s_mount_opt, WRITEBACK_DATA);
 		sbi->s_journal = NULL;
 		needs_recovery = 0;
 		goto no_journal;
